### PR TITLE
MM-18818 Prompt admins to set the Support Email

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Admin Advisor now prompts system admins if _support email_ is not configured.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-18818

#### Release Note

```release-note
The default value of [Support Email](https://docs.mattermost.com/administration/config-settings.html#support-email) (previously _feedback@mattermost.com)_ has been removed.
```

```release-note
Admin Advisor will now prompt system admins about missing configuration for [Support Email](https://docs.mattermost.com/administration/config-settings.html#support-email). It's required to set this value to ensures Mattermost account requests are sent to the correct team for resolution.
```

Webapp PR - https://github.com/mattermost/mattermost-webapp/pull/7774
Docs PR - https://github.com/mattermost/docs/pull/4431

### CODE REVIEW

**Review by mkraft — APPROVED** [review]

**Comment by harshilsharma63** [comment]
@mkraft I've added some tests. Can you please re-approve?

**Review by mkraft — APPROVED** [review]

**Review by sbishel — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]

**Comment by furqanmlk** [comment]
is Space allowed in `Support Email `field ?
I am not getting Bot message with empty space in `Support Email` field

**Comment by harshilsharma63** [comment]
@mkraft @sbishel @furqanmlk I've made a change in this that still notifies sys admins about in-configured support email via Admin Advisor but doesn't send any of the contextual notifications. Request you to re-review.

**Comment by harshilsharma63** [comment]
> Is space allowed in `Support Email `field ?
> I am not getting Bot message with empty space in `Support Email` field

@furqanmlk this has been fixed now. Support email with only whitespace will be considered equivalent to empty now.

File: model/system.go [hunk 1]
```diff
@@ -170,13 +171,21 @@ var WarnMetricsTable = map[string]WarnMetric{
 		IsBotOnly: false,
 		IsRunOnce: true,
 	},
+	SYSTEM_METRIC_SUPPORT_EMAIL_NOT_CONFIGURED: {
+		Id:         SYSTEM_METRIC_SUPPORT_EMAIL_NOT_CONFIGURED,
+		Limit:      -1,
```
Comment (mkraft) [inline model/system.go:176]: Does a `Limit` value of `-1` mean that it'll check every hour for the lifetime of the product?

**Review by sbishel — APPROVED** [review]

File: model/system.go (hunk 1 above)
Comment (harshilsharma63) [reply model/system.go:176]: @mkraft We don't use the `Limit` field for this metric. It's just there because its part of the struct.

Comment (mkraft) [reply model/system.go:176]: Ok. How often will this check run?

Comment (harshilsharma63) [reply model/system.go:176]: It will run each time the server starts, and on top of that once every 7 days.

Comment (mkraft) [reply model/system.go:176]: Ah I see, based on `model.WARN_METRIC_JOB_INTERVAL`. Got it, thanks.

**Review by furqanmlk — APPROVED** [review]
LGTM

**Pull request merged**